### PR TITLE
Add documentation for Transmission component

### PR DESCRIPTION
**Pull request by MatteGary**
Transmission Component

**Description:**
Newly implemented Transmission Component, in order to be able to trigger event for completed and started torrent. Documentation is copied from the previous Transmission Sensor, with the addition of Automation part, configuration and example.
Transmission Sensor is modified in order to point to the component.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant):** home-assistant/home-assistant#19230

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/sensor.transmission.markdown [hunk 1]
```diff
@@ -2,90 +2,16 @@
 layout: page
 title: "Transmission Sensor"
 description: "Instructions on how to integrate Transmission sensors within Home Assistant."
-date: 2015-04-25 9:06
+date: 2018-12-12 22:59
```
Comment (frenck) [inline source/_components/sensor.transmission.markdown]: Please do not update the date. It indicated initial creation.

File: source/_components/sensor.transmission.markdown [hunk 2]
```diff
@@ -2,90 +2,16 @@
...
 sidebar: true
 comments: false
 sharing: true
 footer: true
 logo: transmission.png
 ha_category: Downloading
-ha_release: pre 0.7
+ha_release: 0.85
```
Comment (frenck) [inline source/_components/sensor.transmission.markdown]: Seems incorrect, this sensor is still introduced `pre 0.7`, please revert this change.

File: source/_components/transmission.markdown [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+</p>
+
+## {% linkable_title Configuration %}
+
+To enable this sensor, add the following lines to your `configuration.yaml`:
+
+```yaml
+transmission:
+  host: 192.168.1.1
+  username: user
```
Comment (frenck) [inline source/_components/transmission.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/transmission.markdown [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  password: psw
```
Comment (frenck) [inline source/_components/transmission.markdown]: Same as above.

File: source/_components/transmission.markdown [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  port: 8181
```
Comment (frenck) [inline source/_components/transmission.markdown]: Same as above.

File: source/_components/transmission.markdown [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  monitored_variables:
```
Comment (frenck) [inline source/_components/transmission.markdown]: Same as above.

File: source/_components/transmission.markdown [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+## {% linkable_title Setup %}
+
+To use the monitoring, your transmission client needs to allow remote access. If you are running the graphical transmission client (transmission-gtk) go to **Edit** -> **Preferences** and choose the tab **Remote**. Check **Allow remote access**, enter your username and your password, and uncheck the network restriction as needed.
+
+<p class='img'>
+  <img src='{{site_root}}/images/components/transmission/transmission_perf.png' />
+</p>
+
+If everything is setup correctly, the details will show up in the frontend.
```
Comment (frenck) [inline source/_components/transmission.markdown]:
```suggestion
If everything is set up correctly, the details will show up in the frontend.
```

File: source/_components/transmission.markdown [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+username:
+  description: Your Transmission username, if you use authentication.
+  required: false
+  type: string
+password:
+  description: Your Transmission password, if you use authentication.
+  required: false
+  type: string
+monitored_variables:
+  description: Conditions to display in the frontend. Everyone of them will act as sensor.
```
Comment (frenck) [inline source/_components/transmission.markdown]:
```suggestion
  description: Conditions to display in the frontend. Every one of them will act as a sensor.
```

**Review by frenck — CHANGES_REQUESTED** [review]
@MatteGary Thanks for the PR! I've reviewed it and left you some comments. Could you please take a look? 👍

**Review by frenck — DISMISSED** [review]

**Comment by frenck** [comment]
Thanks for updating the PR @MatteGary 👍 

✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Comment by MatteGary** [comment]
Wondering why I'm no longer an Assignees, since I opened this PR...
btw, the related PR in home-assistant has been approved, so this can be merged.

File: source/_components/transmission.markdown [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+title: "Transmission"
+description: "Instructions on how to integrate Transmission within Home Assistant."
+date: 2018-12-12 22:44
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: transmission.png
+ha_category: Downloading
+ha_release: 0.85
```
Comment (MartinHjelmare) [inline source/_components/transmission.markdown]: 0.87

File: source/_components/transmission.markdown [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+password:
+  description: Your Transmission password, if you use authentication.
+  required: false
+  type: string
+turtle_mode:
+  description: If enabled, it creates a switch entity to control the 'Alternative Speed Limits' (aka 'Turtle mode') setting.
+  required: false
+  type: boolean
+monitored_variables:
+  description: Conditions to display in the frontend. Every one of them will act as a sensor.
```
Comment (MartinHjelmare) [inline source/_components/transmission.markdown]: This doesn't have anything to do with the frontend per se. This sets what conditions will be tracked as entitites.

File: source/_components/transmission.markdown [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+      description: The total number of torrents present in the client.
+    started_torrents:
+      description: The current number of started torrents (downloading).
+    completed_torrents:
+      description: The current number of completed torrents (seeding)
+{% endconfiguration %}
+
+## {% linkable_title Event Automation %}
+
+The Transmission component is continuously monitoring the status of torrents in the target client. Once a torrent is started or completed, an event is triggered on the Home Assistant Bus, allows to implement any kind of automation.
```
Comment (MartinHjelmare) [inline source/_components/transmission.markdown]: an event is triggered on the Home Assistant event bus, which allows to implement any kind of automation.

File: source/_components/transmission.markdown [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+  type: string
+password:
+  description: Your Transmission password, if you use authentication.
+  required: false
+  type: string
+turtle_mode:
+  description: If enabled, it creates a switch entity to control the 'Alternative Speed Limits' (aka 'Turtle mode') setting.
+  required: false
+  type: boolean
+monitored_variables:
```
Comment (kennedyshead) [inline source/_components/transmission.markdown]: This is now `monitored_conditions`

**Comment by MartinHjelmare** [comment]
There are merge conflicts.

**Comment by klaasnicolaas** [comment]
If the deploy give us green light, I think we can merge this PR 😄

**Review by klaasnicolaas — APPROVED** [review]

File: source/_components/transmission.markdown [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+layout: page
+title: "Transmission"
+description: "Instructions on how to integrate Transmission within Home Assistant."
+date: 2018-12-12 22:44
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: transmission.png
+ha_category: Downloading
```
Comment (klaasnicolaas) [inline source/_components/transmission.markdown]:
```suggestion
ha_category: 
  - Downloading
  - Switch
  - Sensor
```

File: source/_components/transmission.markdown [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+ha_release: 0.87
+ha_iot_class: "Local Polling"
+---
```
Comment (klaasnicolaas) [inline source/_components/transmission.markdown:20]:
```suggestion
redirect_from:
  - /components/switch.transmission/
  - /components/sensor.transmission/
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Comment by klaasnicolaas** [comment]
You can remove the sensor and switch pages, if you use my suggestions.

File: source/_components/transmission.markdown [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+  type: string
+password:
+  description: Your Transmission password, if you use authentication.
+  required: false
+  type: string
+turtle_mode:
+  description: If enabled, it creates a switch entity to control the 'Alternative Speed Limits' (aka 'Turtle mode') setting.
+  required: false
+  type: boolean
+monitored_conditions:
```
Comment (klaasnicolaas) [inline source/_components/transmission.markdown:77]:
```suggestion
scan_interval:
  description: How frequently to query for new data. Defaults to 120 seconds.
  required: false
  type: integer
```

Above the `Monitored Conditions`, see also: https://github.com/home-assistant/home-assistant.io/pull/8344

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Comment by MatteGary** [comment]
Everything is ok now?

File: source/_components/transmission.markdown [hunk 16]
```diff
@@ -69,7 +69,10 @@ turtle_mode:
   description: If enabled, it creates a switch entity to control the 'Alternative Speed Limits' (aka 'Turtle mode') setting.
   required: false
   type: boolean
-monitored_conditions:
+scan_interval:
```
Comment (MartinHjelmare) [inline source/_components/transmission.markdown:73]: This isn't an option at the moment. There's an open PR to add it, but that has its own docs PR.

File: source/_components/sensor.transmission.markdown [hunk 17]
```diff
@@ -13,79 +13,5 @@ ha_release: pre 0.7
 ha_iot_class: "Local Polling"
```
Comment (MartinHjelmare) [inline source/_components/sensor.transmission.markdown]: With the redirects we can now remove the sensor and switch pages completely.

File: source/_components/switch.transmission.markdown [hunk 18]
```diff
@@ -13,40 +13,5 @@ ha_release: pre 0.7
 ha_iot_class: "Local Polling"
 ---
 
-The `transmission` switch platform allows you to control your [Transmission](http://www.transmissionbt.com/) client from within Home Assistant. The platform enables you switch to your 'Alternative Speed Limits' (aka 'Turtle mode') setting.
+The `transmission` switch platform allows you to control your [Transmission](http://www.transmissionbt.com/) client from within Home Assistant. The platform enables you to switch to your 'Alternative Speed Limits' (aka 'Turtle mode') setting. To get your Transmission switch working with Home Assistant, follow the instructions for the general [Transmission component](/components/transmission/).
```
Comment (MartinHjelmare) [inline source/_components/switch.transmission.markdown]: See above.

File: source/_components/sensor.transmission.markdown (hunk 17 above)
Comment (MatteGary) [reply source/_components/sensor.transmission.markdown]: Should I removed it from my branch and commit the change?

File: source/_components/transmission.markdown (hunk 16 above)
Comment (MatteGary) [reply source/_components/transmission.markdown:73]: What should I do?

File: source/_components/sensor.transmission.markdown (hunk 17 above)
Comment (klaasnicolaas) [reply source/_components/sensor.transmission.markdown]: Delete the files on your `next-transmission` branch and then commit it to this PR.

File: source/_components/transmission.markdown (hunk 16 above)
Comment (MartinHjelmare) [reply source/_components/transmission.markdown:73]: Remove the `scan_interval` option here.

**Review by MartinHjelmare — DISMISSED** [review]

**Comment by MartinHjelmare** [comment]
🤔 The netlify build preview doesn't find the transmission page. Should we worry?

**Comment by klaasnicolaas** [comment]
@MartinHjelmare Check the deploy of this PR #8344 
What do we do?

**Comment by klaasnicolaas** [comment]
Whoops see that `scan_interval` was consciously deleted already, see what the deploy now does and if the component now appears in the list delete (`scan_interval`) and merge this PR.

**Comment by balloob** [comment]
The issue was that it was missing the `---` after the metadata block, causing Jekyll to think that it was part of the content. Navigating to the url directly showed it: https://deploy-preview-7816--home-assistant-docs.netlify.com/components/transmission/
